scripts/update-message-index.py

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env python3
+
+import argparse
+from itertools import chain
+from pathlib import Path
+from string import Template
+from typing import NamedTuple, Sequence
+import yaml
+import jinja2
+
+
+class Version(NamedTuple):
+    major: int
+    minor: int
+    patch: int = 0
+
+    def __str__(self):
+        return ".".join(map(str, self))
```

Comment:
how about
```python

    @classmethod
    def from_string(raw_version: str) -> "Version":
         version = raw_version.split(".")
         assert 2 <= len(version) <= 3 and all(n.isdecimal() for n in version), raw_version
         return Version(*map(int, version))
```

instead of a free floating function that looks all lonely? :)


Come to think of it, this would also be a good idea for the `parse_entry` and `parse_example` functions.

Reply:
Good idea.